utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,221 @@
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Datatype Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+
+xmlns "urn:hl7-org:v3" as v3;
+
+# Map C-CDA id to FHIR Identifier.
+#
+# + idElement - C-CDA id element
+# + return - Return FHIR Identifier
+public isolated function mapCcdaIdToFhirIdentifier(xml idElement) returns r4:Identifier? {
+    string|error? idVal = idElement.extension;
+    string|error? rootVal = idElement.root;
+    if rootVal is string {
+        if rootVal.equalsIgnoreCaseAscii("2.16.840.1.113883.4.1") && idVal is string {
+            return {
+                system: "http://hl7.org/fhir/sid/us-ssn",
+                value: idVal
+            };
+        } else if idVal is string {
+            return {
+                system: rootVal,
+                value: string `urn:oid:${idVal}`
+            };
+        } else {
+            return {
+                system: "urn:ietf:rfc:3986",
+                value: rootVal
+            };
+        }
+    } else {
+        log:printDebug("Mandatory field root not available", rootVal);
+        return ();
+    }
+}
+
+# Map C-CDA address to FHIR Address.
+#
+# + addressElement - C-CDA address element
+# + return - Return FHIR Address
+public isolated function mapCcdaAddressToFhirAddress(xml addressElement) returns r4:Address? {
+    xml streetAddressLineElement = addressElement/<v3:streetAddressLine|streetAddressLine>;
+    xml cityElement = addressElement/<v3:city|city>;
+    xml stateElement = addressElement/<v3:state|state>;
+    xml postalCodeElement = addressElement/<v3:postalCode|postalCode>;
+
+    string streetAddressLine = streetAddressLineElement.data().trim();
+    string city = cityElement.data().trim();
+    string state = stateElement.data().trim();
+    string postalCode = postalCodeElement.data().trim();
+
+    if streetAddressLine != "" || city != "" || state != "" || postalCode != "" {
+        return {
+            line: [streetAddressLine],
+            city: city,
+            state: state,
+            postalCode: postalCode
+        };
+    } else {
+        log:printDebug("Address fields not available");
+        return ();
+    }
```

Comment:
```suggestion
    }
    log:printDebug("Address fields not available");
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/a482466dc4ba847b3e57e8e27d7d0e8cc48cf07c